hal/shared/cellular_sig_perc_mapping.cpp

```diff
@@ -0,0 +1,225 @@
+    SIG_REGION_GOOD =  3,   /* <= 75%  */
+    SIG_REGION_GREAT = 4,   /* <= 100% */
+    MAX_REGION_THRESHOLDS
+};
+
+/* RF Characterization internal testing. Please see https://docs.google.com/spreadsheets/d/11qpipAc7JWzr8rAk-SOClfhflT-gXabDPGOMQU4be8I/edit?usp=sharing */
+const int CATM1_STRN_MAP[] = {
+    -141,   /* SIG_REGION_POOR */
+    -122,   /* SIG_REGION_BAD */
+    -116,   /* SIG_REGION_FAIR */
+    -107,   /* SIG_REGION_GOOD */
+    -95,    /* SIG_REGION_GREAT */
+    //-44     /* MAX */
+};
+
+const int CATM1_QUAL_MAP[] = {
+    -20,   /* SIG_REGION_POOR */
+    -19,   /* SIG_REGION_BAD */
+    -17,   /* SIG_REGION_FAIR */
+    -14,   /* SIG_REGION_GOOD */
+    -12,   /* SIG_REGION_GREAT */
+    //-3     /* MAX */
+};
+
+/* Source: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/telephony/java/android/telephony/CellSignalStrengthLte.java */
+const int CAT1_STRN_MAP[] = {
+    -141,   /* SIG_REGION_POOR */
+    -115,   /* SIG_REGION_BAD */
+    -105,   /* SIG_REGION_FAIR */
+    -95,    /* SIG_REGION_GOOD */
+    -85,    /* SIG_REGION_GREAT */
+    //-44     /* MAX */
+};
+
+const int CAT1_QUAL_MAP[] = {
+    -20,   /* SIG_REGION_POOR */
+    -19,   /* SIG_REGION_BAD */
+    -17,   /* SIG_REGION_FAIR */
+    -14,   /* SIG_REGION_GOOD */
+    -12,   /* SIG_REGION_GREAT */
+    //-3     /* MAX */
+};
+
+/* Source: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/telephony/java/android/telephony/CellSignalStrengthWcdma.java */
+const int UTRAN_STRN_MAP[] = {
+    -121,   /* SIG_REGION_POOR */
+    -115,   /* SIG_REGION_BAD */
+    -105,   /* SIG_REGION_FAIR */
+    -95,    /* SIG_REGION_GOOD */
+    -85,    /* SIG_REGION_GREAT */
+    //-24     /* MAX */
+};
+
+/* Source: https://comtech.vsb.cz/qualmob/ecno.html */
+const int UTRAN_QUAL_MAP[] = {
+    -24,   /* SIG_REGION_POOR */
+    -17,   /* SIG_REGION_BAD */
+    -10,   /* SIG_REGION_FAIR */
+    -5,    /* SIG_REGION_GOOD */
+    -2,    /* SIG_REGION_GREAT */
+    //0     /* MAX */
+};
+
+/* Source: https://github.com/aosp-mirror/platform_frameworks_base/blob/master/telephony/java/android/telephony/CellSignalStrengthGsm.java */
+const int GSM_STRN_MAP[] = {
+    -111,   /* SIG_REGION_POOR */
+    -107,   /* SIG_REGION_BAD */
+    -103,   /* SIG_REGION_FAIR */
+    -97,    /* SIG_REGION_GOOD */
+    -89,    /* SIG_REGION_GREAT */
+    //-48     /* MAX */
+};
+
+} /* anonymous */
+
+int cellular_get_scaled_strn(int strn_value, hal_net_access_tech_t access_tech) {
+    int res = 0;
+    int region = -1;
+
+    // The strength value obtained is a number multiplied by 100. Get the original value.
+    strn_value = strn_value / 100;
+
+    int gen_strn_map[MAX_REGION_THRESHOLDS] = {0};
+
+    // Get the mapping data according to the RAT
+    switch (access_tech) {
+        case NET_ACCESS_TECHNOLOGY_GSM:
+        case NET_ACCESS_TECHNOLOGY_EDGE: {
+            memcpy(gen_strn_map, GSM_STRN_MAP, sizeof(gen_strn_map));
```

Comment:
Could potentially just set the pointer to the constant array instead of copying.